tests/Order.Service.Tests/MicrocksWebApplicationFactory.cs

```diff
@@ -17,25 +17,39 @@
 
 using System;
 using System.Net;
-using System.Net.Http;
 using System.Net.Sockets;
 using System.Threading.Tasks;
-using DotNet.Testcontainers;
+using System.Threading;
+using System.Linq;
+
 using DotNet.Testcontainers.Builders;
 using DotNet.Testcontainers.Configurations;
+
 using Microcks.Testcontainers;
 using Microcks.Testcontainers.Connection;
+
 using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.Hosting;
+
 using Testcontainers.Kafka;
+
 using Xunit;
 
 namespace Order.Service.Tests;
 
+/// <summary>
+/// Shared WebApplicationFactory for integration tests using Microcks and Kafka containers.
+/// This factory is designed to be used as a singleton across all test classes to optimize container startup time.
+/// Containers are started once and reused by all tests in the test assembly.
+/// </summary>
 public class MicrocksWebApplicationFactory<TProgram> : KestrelWebApplicationFactory<TProgram>, IAsyncLifetime
```

Comment:
I would have named this differently as - IMHO - it's a more generic concern and shouldn't be related to Microcks. Maybe simply `OrderServiceWebApplicationFactory`?